google/cloud/forseti/notifier/notifier.py

```diff
@@ -74,58 +75,61 @@ def convert_to_timestamp(violations):
     return violations
 
 
-def process(message):
-    """Process messages about what notifications to send.
+def inv_summary_notify(inv_index_id, service_config):
+    """Emit an inventory summary notification if/as needed.
 
     Args:
-        message (dict): Message with payload in dict.
-            The payload will be different depending on the sender
-            of the message.
-
-            Example:
-                {'status': 'foobar_done',
-                 'payload': {}}
+        inv_index_id (str): Inventory index id.
+        service_config (ServiceConfig): Forseti 2.0 service configs.
     """
-    payload = message.get('payload')
-
-    if message.get('status') == 'inventory_done':
-        inv_email_notifier = inv_summary.EmailInventorySnapshotSummary(
-            payload.get('sendgrid_api_key')
-        )
-        inv_email_notifier.run(
-            payload.get('cycle_time'),
-            payload.get('cycle_timestamp'),
-            payload.get('snapshot_cycle_status'),
-            payload.get('notifiers'),
-            payload.get('email_sender'),
-            payload.get('email_recipient')
-        )
+    notifier_config = service_config.get_notifier_config()
+    if not notifier_config.get('inventory'):
+        LOGGER.info('No "inventory" configuration for notifier.')
+        return
+
+    if not notifier_config.get('inventory').get('summary'):
+        LOGGER.info('No "inventory summary" configuration for notifier.')
+        return
+
+    notifier_config = notifier_config.get('inventory').get('summary')
+
+    if not notifier_config.get('enabled'):
+        LOGGER.info('Inventory summary notifications are turned off.')
         return
 
-    if message.get('status') == 'scanner_done':
-        scanner_email_notifier = scanner_summary.EmailScannerSummary(
-            payload.get('sendgrid_api_key')
-        )
-        scanner_email_notifier.run(
-            payload.get('output_csv_name'),
-            payload.get('output_filename'),
-            payload.get('now_utc'),
-            payload.get('all_violations'),
-            payload.get('resource_counts'),
-            payload.get('violation_errors'),
-            payload.get('email_sender'),
-            payload.get('email_recipient'),
-            payload.get('email_description'))
+    if not notifier_config.get('gcs_path'):
+        LOGGER.error('"gcs_path" not set for inventory summary notifier.')
         return
 
+    with service_config.scoped_session() as session:
+        inv_index = session.query(InventoryIndex).get(inv_index_id)
+        if inv_index.notified_at_datetime:
+            LOGGER.info(
+                'Inventory summary notification already sent (%s).',
+                inv_index.notified_at_datetime)
+            return
+
+        summary_data = inv_index.get_summary(session)
```

Comment:
Just a comment, and not a TODO: it might be worth considering to save the summary data to the Inventory Index table, after each crawler run as it seems useful outside of notification.

Reply:
Added issue #1427.